### PR TITLE
Directize: Fix handling of non-nullable locals and unreachability

### PR DESCRIPTION
**Pull request by kripken**
The order of operations could allow us to add vars but then later decide
not to do the optimization due to unreachability. And then we did not do a
fixup for non-nullability for those args, leading to a fuzzer error.

### CODE REVIEW

File: test/lit/passes/directize_all-features.wast [hunk 1]
```diff
@@ -807,6 +808,52 @@
...
+ ;; CHECK-NEXT:  (call_indirect $0 (type $F)
+ ;; CHECK-NEXT:   (unreachable)
+ ;; CHECK-NEXT:   (select
+ ;; CHECK-NEXT:    (i32.const 10)
+ ;; CHECK-NEXT:    (i32.const 11)
+ ;; CHECK-NEXT:    (local.get $x)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $select-non-nullable-unreachable-arg (param $x i32)
```
Comment (aheejin) [inline test/lit/passes/directize_all-features.wast:845]: Did we do anything for this in this PR?

**Review by aheejin — APPROVED** [review]

File: test/lit/passes/directize_all-features.wast (hunk 1 above)
Comment (kripken) [reply test/lit/passes/directize_all-features.wast:845]: Technically the PR didn't do anything for it, correct - this wasn't broken before (since we checked unreachability before adding any vars). Still, for completeness this test means we check all the corner cases in this area to verify we never create a var if we don't optimize.
